contracts/v2/ColabBankV2.sol

```diff
@@ -18,7 +18,7 @@ contract ColabBankV2 {
     function deposit() public payable {
         require(msg.value != 0, "cannot deposit 0 ETH");
         uint256 value = msg.value;
-        balances[msg.sender] = value;
+        balances[msg.sender] += value;
```

Comment:
I'd suggest you finalize `ColabBank` first before proposing changes to `ColabBankV2`. nice attempt tho